### PR TITLE
resolve #9102

### LINKED ISSUE #9102: [bug] When IdentityClients set more then one(Default),HttpClientAuthenticator will fail
* Your **ABP Framework version**.
4.3.0
* Your **User Interface** type (Angular/MVC/React... etc.) if the issue is related to a specific UI
don't care
* Your database provider(EF Core/MongoDB)
don't care
* Exception message and **stack trace** if available (check the logs).
`Volo.Abp.Http.Client.AbpRemoteCallException: 'Unauthorized'`
* Steps needed to **reproduce** the problem.
For test this problem,I use `MyCompanyName.MyProjectName.HttpApi.Client.ConsoleTestApp` in `module` of `template`:
1:`update-database` for `MyCompanyName.MyProjectName.IdentityServer`
2.Run `MyCompanyName.MyProjectName.HttpApi.Host` and `MyCompanyName.MyProjectName.IdentityServer`
3.modify `appsettings.json` in `MyCompanyName.MyProjectName.HttpApi.Client.ConsoleTestApp`like:
```json
{
  "RemoteServices": {
    "Default": {
      "BaseUrl": "https://localhost:44301/"
    },
    "MyProjectName": {
      "BaseUrl": "https://localhost:44300/"
    }
  },
  "IdentityClients": {
    "MyProjectName": {
      "GrantType": "password",
      "ClientId": "MyProjectName_App",
      "ClientSecret": "1q2w3e*",
      "UserName": "admin",
      "UserPassword": "1q2w3E*",
      "Authority": "https://localhost:44301/",
      "Scope": "MyProjectName"
    }
  }
}
```
4.Run `MyCompanyName.MyProjectName.HttpApi.Client.ConsoleTestApp` will receive error
image
5.After check source code,I think something wrong with `IdentityModelRemoteServiceHttpClientAuthenticator` 
```csharp
 public class IdentityModelRemoteServiceHttpClientAuthenticator : IRemoteServiceHttpClientAuthenticator, ITransientDependency
    {
        protected IIdentityModelAuthenticationService IdentityModelAuthenticationService { get; }

        public IdentityModelRemoteServiceHttpClientAuthenticator(
            IIdentityModelAuthenticationService identityModelAuthenticationService)
        {
            IdentityModelAuthenticationService = identityModelAuthenticationService;
        }

        public virtual async Task Authenticate(RemoteServiceHttpClientAuthenticateContext context)
        {
            await IdentityModelAuthenticationService.TryAuthenticateAsync(
                context.Client,
                context.RemoteService.GetIdentityClient()//context.RemoteServiceName
            );
        }
    }
```
If I modify `context.RemoteService.GetIdentityClient()` to `context.RemoteServiceName`,It's working fine
Could you check it ,thanks

### PR DESCRIPTION
**Pull request by yinchang0626**

### CODE REVIEW

File: framework/src/Volo.Abp.IdentityModel/Volo/Abp/IdentityModel/IdentityModelAuthenticationService.cs [hunk 1]
```diff
@@ -255,7 +255,7 @@ protected virtual string CalculateDiscoveryDocumentCacheKey(IdentityClientConfig
 
         protected virtual string CalculateTokenCacheKey(IdentityClientConfiguration configuration)
         {
-            return IdentityModelTokenCacheItem.CalculateCacheKey(configuration);
+            return (IdentityModelTokenCacheItem.CalculateCacheKey(configuration) + "," + CurrentTenant.Id).ToMd5();
```
Comment (maliming) [inline framework/src/Volo.Abp.IdentityModel/Volo/Abp/IdentityModel/IdentityModelAuthenticationService.cs]: https://github.com/abpframework/abp/issues/9112#issuecomment-846714489

File: framework/src/Volo.Abp.Http.Client.IdentityModel/Volo/Abp/Http/Client/IdentityModel/IdentityModelRemoteServiceHttpClientAuthenticator.cs [hunk 2]
```diff
@@ -20,7 +20,7 @@ public virtual async Task Authenticate(RemoteServiceHttpClientAuthenticateContex
         {
             await IdentityModelAuthenticationService.TryAuthenticateAsync(
                 context.Client,
-                context.RemoteService.GetIdentityClient()
+                context.RemoteServiceName
```
Comment (maliming) [inline framework/src/Volo.Abp.Http.Client.IdentityModel/Volo/Abp/Http/Client/IdentityModel/IdentityModelRemoteServiceHttpClientAuthenticator.cs]: How about `context.RemoteService.GetIdentityClient() ?? context.RemoteServiceName`?

Comment (yinchang0626) [reply framework/src/Volo.Abp.Http.Client.IdentityModel/Volo/Abp/Http/Client/IdentityModel/IdentityModelRemoteServiceHttpClientAuthenticator.cs]: @maliming Hi:
I though code inside  `context.RemoteService.GetIdentityClient()`  is wrong
```csharp
    public static class RemoteServiceConfigurationExtensions
    {
        public const string IdentityClientName = "IdentityClient";
        public const string UseCurrentAccessTokenName = "UseCurrentAccessToken";

        [CanBeNull]
        public static string GetIdentityClient([NotNull] this RemoteServiceConfiguration configuration)
        {
            Check.NotNullOrEmpty(configuration, nameof(configuration));

            return configuration.GetOrDefault(IdentityClientName);
        }
}
public class RemoteServiceConfiguration : Dictionary<string, string>
    {
        /// <summary>
        /// Base Url.
        /// </summary>
        public string BaseUrl
        {
            get => this.GetOrDefault(nameof(BaseUrl));
            set => this[nameof(BaseUrl)] = value;
        }

        /// <summary>
        /// Version.
        /// </summary>
        public string Version
        {
            get => this.GetOrDefault(nameof(Version));
            set => this[nameof(Version)] = value;
        }

        public RemoteServiceConfiguration()
        {
            
        }

        public RemoteServiceConfiguration(string baseUrl, string version = null)
        {
            this[nameof(BaseUrl)] = baseUrl;
            this[nameof(Version)] = version;
        }
    }
```
 `RemoteServiceConfiguration` seems did't has `IdentityClient` key,
So,I don't know why use the  `context.RemoteService.GetIdentityClient()` method to get `RemoteServiceName`,Maybe we should check with @hikalkan 
On the other hand,
should we define `appsetting.json` like below?(I did't test it)
```json
{
  "RemoteServices": {
    "Default": {
      "BaseUrl": "https://localhost:44301/"
    },
    "MyProjectName": {
      "BaseUrl": "https://localhost:44300/",
      "IdentityClients":"MyProjectName"
    }
  },
  "IdentityClients": {
    "Default": {
      "GrantType": "password",
      "ClientId": "MyProjectName_App",
      "ClientSecret": "1q2w3e*",
      "UserName": "admin",
      "UserPassword": "1q2w3E*",
      "Authority": "https://localhost:44301/",
      "Scope": "MyProjectName"
    },
    "MyProjectName": {
      "GrantType": "password",
      "ClientId": "MyProjectName_App",
      "ClientSecret": "1q2w3e*",
      "UserName": "admin",
      "UserPassword": "1q2w3E*",
      "Authority": "https://localhost:44301/",
      "Scope": "MyProjectName"
    }
  }
}
```

**Comment by yinchang0626** [comment]
@maliming Hi
I modified code by following your comment,and also double check it
1.`[IgnoreMultiTenancy]` remove in `IdentityModelTokenCacheItem` 
2.`context.RemoteService.GetIdentityClient() ?? context.RemoteServiceName` in `IdentityModelAuthenticationService.cs `
So ,both of below `appsettings.json` works fine.Thanks
```json
{
  "RemoteServices": {
    "Default": {
      "BaseUrl": "https://localhost:44301/"
    },
    "MyProjectName": {
      "BaseUrl": "https://localhost:44300/",
     "IdentityClient":"MyProjectName"
    }
  },
  "IdentityClients": {
    "Default": {
      "GrantType": "password",
      "ClientId": "MyProjectName_App",
      "ClientSecret": "1q2w3e*",
      "UserName": "admin",
      "UserPassword": "1q2w3E*",
      "Authority": "https://localhost:44301/",
      "Scope": "MyProjectName"
    },
    "MyProjectName": {
      "GrantType": "password",
      "ClientId": "MyProjectName_App",
      "ClientSecret": "1q2w3e*",
      "UserName": "admin",
      "UserPassword": "1q2w3E*",
      "Authority": "https://localhost:44301/",
      "Scope": "MyProjectName"
    }
  }
}
```
```json
{
  "RemoteServices": {
    "Default": {
      "BaseUrl": "https://localhost:44301/"
    },
    "MyProjectName": {
      "BaseUrl": "https://localhost:44300/"
    }
  },
  "IdentityClients": {
    "MyProjectName": {
      "GrantType": "password",
      "ClientId": "MyProjectName_App",
      "ClientSecret": "1q2w3e*",
      "UserName": "admin",
      "UserPassword": "1q2w3E*",
      "Authority": "https://localhost:44301/",
      "Scope": "MyProjectName"
    }
  }
}
```

**Comment by maliming** [comment]
Thanks @yinchang0626 

Can you open PR in the `rel-4.3` branch?

**Comment by yinchang0626** [comment]
@maliming 
OK, I open a new P.R. base 4.3-rel at #9151 
and you can close this P.R

**Pull request closed without merging**